api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -91,6 +94,7 @@
       UserService userService,
       UserRecentResourceService userRecentResourceService) {
     this.workspaceService = workspaceService;
+    this.workspaceDao = workspaceDao;
```

Comment:
tl;dr: Should just call `workspaceService.getDao()` where needed for consistency with the rest of the file.

Looking again at the rest of this controller - this is probably a broader issue.

Arguably it might be more consistent for access to go through a new method in WorkspaceService instead of straight to workspaceDao. I don't see clear rationale for why we'd use workspaceService for some things, and workspaceDao directly for others.

The existing code has this same issue by just calling `getDao()` on the service, which equally broke the abstraction.

Reply:
Yeah, okay. I totally missed that `getDao()` existed :( 
I think we call directly to the Dao when it is simple, and the Service when there is more complicated logic around what to do.